### ISSUE TITLE
[API] Authentication 

### ISSUE THREAD

**Issue by nlamirault** [pi-hole/AdminLTE#483]
**In raising this issue, I confirm the following (please check boxes, eg [X] - no spaces) Failure to fill the template will close your issue:**

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*
- [x] The issue I'm reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the codebase?:**
1

---
**[QUESTION]:**

With a simple curl for the API, i haven't got topClients, top_queries, ... results : 

```bash
$ curl "http://192.168.1.23:9180/admin/api.php?summaryRaw&overTimeData&topItems&recentItems&getQueryTypes&getForwardDestinations&getQuerySources&jsonForceObject"

{"domains_being_blocked":116759,"dns_queries_today":37594,"ads_blocked_today":128,"ads_percentage_today":0.34047986380805,"domains_over_time":{},"ads_over_time":{"1":0}}
```
In a browser, after login into, 

```bash
http://192.168.1.23:9180/admin/api.php?summaryRaw&overTimeData&topItems&recentItems&getQueryTypes&getForwardDestinations&getQuerySources&jsonForceObject
{"domains_being_blocked":116759,"dns_queries_today":37589,"ads_blocked_today":128,"ads_percentage_today":0.34052515363537,"domains_over_time":{},"ads_over_time":{"1":0},"top_queries":{"twitter.com":1174,"www.google.com":981,"www.google.fr":928,"clients3.google.com":885,"mtalk.google.com":640,"api.twitter.com":630,"t.co":467,"www.googleapis.com":459,"connectivitycheck.android.com":417,"query.hicloud.com":356},"top_ads":{"ping.chartbeat.net":236,"t.teads.tv":192,"sync.teads.tv":192,"a.teads.tv":182,"bttrack.com":180,"x.bidswitch.net":170,"dis.criteo.com":158,"static.criteo.net":142,"w.estat.com":140,"gum.criteo.com":126},"query[A]":8714,"query[AAAA]":28875,"8.8.4.4":16165,"8.8.8.8":9343,"top_sources":{"192.168.1.14":37000,"192.168.1.18":492,"192.168.1.25":95,"192.168.1.19":2}}

```

Is there any API to do authentication and have more informations into response ?
Thanks

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

**Comment by donmahallem** [comment]
Afaik you will have to send a form post request with the password in the "pw" field and save the returned cookie and send it with every request.

**Comment by DL6ER** [comment]
> Is there any API to do authentication and have more informations into response ?

Yes, albeit I'm not sure if this is already part of the current `master` version or only of the soon to be released Pi-hole `v3.0`. Appending

```
&auth=YOURPASSWORDTOKEN
```

to your request should do what you want. It could look like, e.g.

```
curl "http://192.168.1.23:9180/admin/api.php?summaryRaw&overTimeData&topItems&recentItems&getQueryTypes&getForwardDestinations&getQuerySources&jsonForceObject&auth=YOURPASSWORDTOKEN"
```

where you can get this mysterious `YOURPASSWORDTOKEN` from your `/etc/pihole/setupVars.conf` file.

**Comment by donmahallem** [comment]
Or to add to @DL6ER 's answer you could instead of copying the token from the conf file create the token from your admin password as seen in [password.php](https://github.com/pi-hole/AdminLTE/blob/d7fdaf9273ad0a3f352bc62d8532330c56b915aa/scripts/pi-hole/php/password.php#L39).
